backend/src/packages/chaiNNer_standard/image/create_images/text_as_image.py

```diff
@@ -0,0 +1,212 @@
+
+
+TEXT_AS_IMAGE_POSITION_LABELS = {
+    TextAsImagePosition.TOP_LEFT: "Top left",
+    TextAsImagePosition.TOP_CENTERED: "Top centered",
+    TextAsImagePosition.TOP_RIGHT: "Top right",
+    TextAsImagePosition.CENTERED_LEFT: "Centered left",
+    TextAsImagePosition.CENTERED: "Centered",
+    TextAsImagePosition.CENTERED_RIGHT: "Centered right",
+    TextAsImagePosition.BOTTOM_LEFT: "Bottom left",
+    TextAsImagePosition.BOTTOM_CENTERED: "Bottom centered",
+    TextAsImagePosition.BOTTOM_RIGHT: "Bottom right",
+}
+
+TEXT_AS_IMAGE_X_Y_REF_FACTORS = {
+    TextAsImagePosition.TOP_LEFT: {"x": np.array([0, 0.5]), "y": np.array([0, 0.5])},
+    TextAsImagePosition.TOP_CENTERED: {
+        "x": np.array([0.5, 0]),
+        "y": np.array([0, 0.5]),
+    },
+    TextAsImagePosition.TOP_RIGHT: {"x": np.array([1, -0.5]), "y": np.array([0, 0.5])},
+    TextAsImagePosition.CENTERED_LEFT: {
+        "x": np.array([0, 0.5]),
+        "y": np.array([0.5, 0]),
+    },
+    TextAsImagePosition.CENTERED: {"x": np.array([0.5, 0]), "y": np.array([0.5, 0])},
+    TextAsImagePosition.CENTERED_RIGHT: {
+        "x": np.array([1, -0.5]),
+        "y": np.array([0.5, 0]),
+    },
+    TextAsImagePosition.BOTTOM_LEFT: {
+        "x": np.array([0, 0.5]),
+        "y": np.array([1, -0.5]),
+    },
+    TextAsImagePosition.BOTTOM_CENTERED: {
+        "x": np.array([0.5, 0]),
+        "y": np.array([1, -0.5]),
+    },
+    TextAsImagePosition.BOTTOM_RIGHT: {
+        "x": np.array([1, -0.5]),
+        "y": np.array([1, -0.5]),
+    },
+}
+
+
+@create_images_group.register(
+    schema_id="chainner:image:text_as_image",
+    name="Text As Image",
+    description="Create an image using any text.",
+    icon="MdTextFields",
+    inputs=[
+        TextInput("Text", multiline=True),
+        BoolInput("Bold", default=False),
+        BoolInput("Italic", default=False),
+        ColorInput(channels=[1, 3], default=Color.bgr((0, 0, 0))),
+        EnumInput(
+            TextAsImageAlignment,
+            label="Text alignment",
+            option_labels=TEXT_AS_IMAGE_ALIGNMENT_LABELS,
+            default=TextAsImageAlignment.CENTERED,
+        ),
+        NumberInput(
+            "Width",
+            minimum=1,
+            maximum=None,
+            controls_step=1,
+            precision=0,
+            default=100,
+        ),
+        NumberInput(
+            "Height",
+            minimum=1,
+            maximum=None,
+            controls_step=1,
+            precision=0,
+            default=100,
+        ),
+        EnumInput(
+            TextAsImagePosition,
+            label="Position",
+            option_labels=TEXT_AS_IMAGE_POSITION_LABELS,
+            default=TextAsImagePosition.CENTERED,
+        ),
+    ],
+    outputs=[
+        ImageOutput(
+            image_type="""
+                Image {
+                    width: Input5 & uint,
+                    height: Input6 & uint,
+                    channels: Input3.channels
```

Comment:
I think this is wrong. The color is either grayscale or RGB, but the image is either grayscale or RGBA. So you probably meant to do `match Input3.channels { 1 => 1, 3 => 4 }`.

Reply:
Finally, decided to output a RGBA image only.
I see no reason to create a grayscale image without transparent layer. If needed, user can use the "change colorspace" node.

Reply:
that makes sense to me